src/GeometryInstancing/Program.cs

```diff
@@ -0,0 +1,261 @@
+                DepthStencilState = new DepthStencilStateDescription(
+                    depthTestEnabled: true,
+                    depthWriteEnabled: true,
+                    comparisonKind: ComparisonKind.LessEqual),
+                RasterizerState = new RasterizerStateDescription(
+                    cullMode: FaceCullMode.Back,
+                    fillMode: PolygonFillMode.Solid,
+                    frontFace: FrontFace.Clockwise,
+                    depthClipEnabled: true,
+                    scissorTestEnabled: false
+                ),
+                PrimitiveTopology = PrimitiveTopology.TriangleStrip,
+                ResourceLayouts = new ResourceLayout[] {_resourceLayout},
+                ShaderSet = new ShaderSetDescription(
+                    // The ordering of layouts directly impacts shader layout schemes
+                    vertexLayouts: new VertexLayoutDescription[] {vertexLayout,vertexLayoutPerInstance},
+                    shaders: new Shader[] {_vertexShader,_fragmentShader}
+                ),
+                Outputs = _graphicsDevice.SwapchainFramebuffer.OutputDescription
+            };
+
+            _pipeline = _factory.CreateGraphicsPipeline(pipelineDescription);
+
+            _commandList = _factory.CreateCommandList();
+
+        }
+
+        private static Shader LoadShader(ShaderStages stage)
+        {
+            string extension = null;
+            switch (_graphicsDevice.BackendType)
+            {
+                case GraphicsBackend.Direct3D11:
+                    extension = "hlsl.bytes";
+                    // not implemented
+                    throw new System.InvalidOperationException();
+                    //break;
+                case GraphicsBackend.Vulkan:
+                    extension = "spv";
+                    // not implemented
+                    throw new System.InvalidOperationException();
+                    //break;
+                case GraphicsBackend.OpenGL:
+                    extension = "glsl";
+                    break;
+                case GraphicsBackend.Metal:
+                    extension = "metallib";
+                    break;
+                default: throw new System.InvalidOperationException();
+            }
+
+            string entryPoint = stage == ShaderStages.Vertex ? "VS" : "FS";
+            string path = Path.Combine(System.AppContext.BaseDirectory, "Shaders", $"{stage.ToString()}.{extension}");
+            byte[] shaderBytes = File.ReadAllBytes(path);
+            return _graphicsDevice.ResourceFactory.CreateShader(new ShaderDescription(stage, shaderBytes, entryPoint));
+        }
+
+        private static void Draw()
+        {
+            // Begin() must be called before commands can be issued.
+            _commandList.Begin();
+
+            // We want to render directly to the output window.
+            _commandList.SetFramebuffer(_graphicsDevice.SwapchainFramebuffer);
+            _commandList.SetFullViewports();

```

Comment:
This automatically gets called when you call `SetFramebuffer`, so feel free to remove it. I know some of the other samples do it -- I made the change fairly late before the release and didn't fix everywhere up.

Reply:
Do you mean SetFullViewports?

Reply:
Yeah, sorry for not being clear. I found that 99% of the time, I was calling `SetFullViewports` after binding a Framebuffer, so I just made it automatic. It can be removed from here.

Reply:
See https://github.com/mellinoe/veldrid/blob/master/src/Veldrid/CommandList.cs#L179-L180